src/edu/csus/ecs/pc2/ui/AccountsTablePane.java

```diff
@@ -0,0 +1,932 @@
+     * 
+     */
+    private void initialize() {
+        this.setLayout(new BorderLayout());
+        this.setSize(new java.awt.Dimension(625, 216));
+        this.add(getMessagePanel(), java.awt.BorderLayout.NORTH);
+        this.add(getButtonPane(), java.awt.BorderLayout.SOUTH);
+        this.add(getScrollPane(), java.awt.BorderLayout.CENTER);
+    }
+
+    @Override
+    public String getPluginTitle() {
+        return "Accounts Table Pane";
+    }
+
+    protected Object[] buildAccountRow(Account account) {
+//      Object[] cols = { "Site", "Type", "Account Id", "Display Name" , "Group", "Alias", "External ID", "ClientId" };
+
+        try {
+            int cols = accountTableModel.getColumnCount();
+            Object[] s = new Object[cols];
+
+            ClientId clientId = account.getClientId();
+            s[0] = getSiteTitle("" + account.getSiteNumber());
+            s[1] = clientId.getClientType().toString().toLowerCase();
+            s[2] = "" + clientId.getClientNumber();
+
+            s[3] = getTeamDisplayName(account);
+            s[4] = getGroupName(account);
+            s[5] = getTeamAlias(account);
+            s[6] = getExternalId(account);
+            // This column is invisible for the "unique" ID
+            s[7] = clientId;
+            return s;
+        } catch (Exception exception) {
+            StaticLog.getLog().log(Log.INFO, "Exception in buildAccountRow()", exception);
+        }
+        return null;
+    }
+
+    private String getGroupName(Account account) {
+        String groupName = "";
+        if (account.getGroupId() != null) {
+            Group group = getContest().getGroup(account.getGroupId());
+            if (group != null) {
+                groupName = group.getDisplayName();
+            }
+        }
+        return groupName;
+    }
+
+    private String getSiteTitle(String string) {
+        return "Site " + string;
+    }
+
+    private String getTeamDisplayName(Account account) {
+        if (account != null) {
+            return account.getDisplayName();
+        }
+
+        return "Invalid Account";
+    }
+    
+
+    private String getTeamAlias(Account account) {
+        if (account != null) {
+            String alias = account.getAliasName();
+            if (alias != null) {
+                return alias;
+            }
+        }
+
+        return "";
+    }
+
+    private String getExternalId(Account account) {
+        if (account != null) {
+            String eid = account.getExternalId();
+            if (eid != null) {
+                return eid;
+            }
+        }
+
+        return "";
+    }
+
+    /**
+     * This method initializes accountTable and accountTableModel
+     * 
+     * @return JTableCustomized
+     */
+    private JTableCustomized getAccountsTable() {
+        if (accountTable == null) {
+            int i;
+            Object[] cols = { "Site", "Type", "Account Id", "Display Name" , "Group", "Alias", "External Id", "ClientId"};
```

Comment:
Ok, to be clear, the ClientId column is not displayed, as it is an actual ClientId object (to indentify the row internally).  The "External Id" is the column is, as you said the CMD Id.  Is "External Id" sufficient or should I change that to CMS Id ?

Reply:
Oh... the ClientId class fields are already on the table,  ClientId = Site + Client Type + Client Number.  So that
information is already present.
That is why I thought it was the ClientId column was something more, a CMS Id, aka Account's getExternalId().

Reply:
Since adding columns is out of scope, removing that column should be done.
If a new column is to be added a CMS Id would be a good choice.

Reply:
ClientId is an internal only column used to lookup rows in a unique way.  This is needed because sorting changes the row<->account mapping (The old account tab did this too, but the MCLB had a special mechanism to identify a record: keys).  Since JTable doesn't have that, we just use an invisible column to the JTable (visual), but it is present in the TableModel (data).

ClientId is required.  I put it in the list of column headers so we know what it is and it gets the column count correct for the TableModel (data).

I did add "External Id", which is visible and IS the CMS Id since, well, it was a minor (harmless?) thing to add and makes the Accounts table "better" (imho).  I always wondered why there was never a way to see this, short of generating a report.  Our RCD likes to see this to compare with teams on the ICPC database website.  I also like to see it, since I generate the results.tsv file, and would like to make sure we have correct CMS Id's BEFORE generating the file.  I can easily remove it in the version for the PR (but I'll keep it my private PC2 distribution for our regional).  That's up to you folks.

Reply:
Exactly, especially if someone has a desire/use for a new column, add it.
Add the CMS Id column - thanks.
Remember to add that new requirement to the PR.   At this time we use the PR to
identify what to test.

Reply:
Changed.